### ISSUE TITLE
Question about DataParallel

### ISSUE THREAD

**Issue by tom-bird** [pytorch/xla#2076]
## ❓ Questions and Help

I have a simple question about the [DataParallel](https://github.com/pytorch/xla/blob/2b2085a290cb3694fa1bda892f9cfe74317c7a17/torch_xla/distributed/data_parallel.py#L36) class. I am running the [test_train_cifar script](https://github.com/pytorch/xla/blob/2b2085a290cb3694fa1bda892f9cfe74317c7a17/test/test_train_cifar.py) on a cloud TPU node, so with 8 TPU devices.

How does the optimization work with these 8 TPU devices? It seems like each of the 8 models on each of the 8 devices performs it's own gradient updates for an entire epoch, without communication, since the `train_loop_fn` iterates through the entire data loader. The data loader on each device will only have 1/8th of the full train set since it is split across the devices. But still, doesn't there need to be some communication between devices in an epoch? Otherwise we are performing very asynchronous SGD.

**Comment by ultrons** [comment]
You can look into `xm.optimizer_step(optimizer)` call. It performs the all_reduce with the gradients calculated on individual data slice by each core, and then optimizer step, i.e. parameter updates.

**Comment by tom-bird** [comment]
Ah I see, thanks!
